core/src/auctions.ts

```diff
@@ -67,26 +67,50 @@ const calculateCollateralToCoverDebt = async function (network: string, auction:
     return auction.debtDAI.dividedBy(marketPriceForAllCollateral);
 };
 
-const enrichAuctionWithMarketValues = async function (auction: Auction, network: string): Promise<Auction> {
+export const enrichAuctionWithMarketData = async function (auction: Auction, network: string): Promise<Auction> {
     if (!auction.isActive || !auction.approximateUnitPrice || auction.isFinished) {
         return auction;
     }
     try {
         const collateralToCoverDebt = await calculateCollateralToCoverDebt(network, auction);
-        const marketUnitPrice = await getMarketPrice(network, auction.collateralSymbol, collateralToCoverDebt);
-        const marketUnitPriceToUnitPriceRatio = auction.approximateUnitPrice
-            .minus(marketUnitPrice)
-            .dividedBy(marketUnitPrice);
-        const auctionWithMarketValues = {
+        let marketDataRecords = await getMarketData(network, auction.collateralSymbol, collateralToCoverDebt);
+        const exchangeFees = await getDefaultMarketFee(network);
+        const exchangeFeePerUnitDAI = exchangeFees.exchangeFeeDAI.dividedBy(collateralToCoverDebt);
+        for (let marketData of Object.values(marketDataRecords)) {
```

Comment:
Currently, you put the whole loop into the try/catch here. So in case any market fails (which is expected, see comment below `catch`) the auction will be returned without any market data. Please make so that each market data is checked separately

Nit: you _might_ want to move the whole loop into a separate function to simplify try/catching it.

Reply:
So the question then boils down to: why do you need try catch above the whole loop then if you catch market problems on the level deeper? As it will return incorrect state, where `marketDataRecords === undefined`.

The proposed solution is to wrap the parts into try/catch at the place where you need to replace them with other "error"-like objects

Reply:
The whole thing is still in the try/catch, right?